packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4186,18 +4202,28 @@ class InheritedElement extends ProxyElement {
   /// Calls [Element.didChangeDependencies] of all dependent elements, if
   /// [InheritedWidget.updateShouldNotify] returns true.
   ///
-  /// Notifies all dependent elements that this inherited widget has changed.
+  /// Called by [update], immediately prior to [build].
   ///
-  /// [InheritedElement] calls this function if the [widget]'s
-  /// [InheritedWidget.updateShouldNotify] returns true.
+  /// Calls [notifyClients] to actually trigger the notifications.
+  @override
+  void updated(InheritedWidget oldWidget) {
+    if (widget.updateShouldNotify(oldWidget))
```

Comment:
I suppose `updateShouldNotify` is now a bit of a misnomer, since notifyClients is only called by default.  I'm not suggesting that the method should be called `updateShouldCallUpdated` though.

Reply:
I think from the perspective of people using the API, it's still conceptually more useful to think of it as a method that says whether to send the notifications.